### PR TITLE
Register OVN-IC APIs

### LINKED ISSUE #2691: Update submariner-operator to register OVN-IC apis
Update submariner-operator to register OVN-IC apis

### PR DESCRIPTION
**Pull request by Jaanki**
Epic: https://github.com/submariner-io/enhancements/issues/186

### CODE REVIEW

**Comment by aswinsuryan** [comment]
@Jaanki Thanks for the PR , I tested it and it works fine.

File: pkg/gateway/crds.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 
 // Ensure ensures that the required resources are deployed on the target system.
 // The resources handled here are the gateway CRDs: Cluster and Endpoint.
+//nolint:gocyclo // No further refactors necessary
```
Comment (aswinsuryan) [inline pkg/gateway/crds.go:33]: a space is needed here after // ?

Comment (Jaanki) [reply pkg/gateway/crds.go:33]: It complains about gci error but running `gci write` on the file doesnot make any changes.

Comment (skitt) [reply pkg/gateway/crds.go:33]: `nolint` has to be introduced by `//` with no space, the explanatory comment needs a space. So this is fine as-is.

**Review by skitt — APPROVED** [review]

**Review by vthapar — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**